File: 3D/src/kmerrill285/PixelInventor/game/client/rendering/textures/Textures.java
package kmerrill285.PixelInventor.game.client.rendering.textures;

public class Textures {
	
	public static Texture TILE_SELECTION = loadTexture("PixelInventor", "gui/tile_selection");
	public static Texture FP_CURSOR = loadTexture("PixelInventor", "gui/fp_cursor");
	public static Texture VIGINETTE = loadTexture("PixelInventor", "gui/viginette");
	public static Texture HOTBAR = loadTexture("PixelInventor", "gui/hotbar");
	public static Texture INVENTORY = loadTexture("PixelInventor", "gui/inventory");
	public static Texture HOTBAR_SELECTED = loadTexture("PixelInventor", "gui/hotbar_selected");
	public static Texture INVENTORY_SELECTED = loadTexture("PixelInventor", "gui/inventory_selected");
	public static Texture WHITE_SQUARE = loadTexture("PixelInventor", "gui/white_square");

	
	public static TextureAtlas TILES = new TextureAtlas("PixelInventor", "tiles");
	
	public static String MINING_LOCATION = "PixelInventor:mining";
	
	public static Texture loadTexture(String modId, String texture) {
		return new Texture(modId, "textures/"+texture+".png");
	}
	
	public static void load() {
//		TILES.addToImage("PixelInventor","gui","mining");
	}
}

File: MonoGame/source/Graphics/Textures/Textures.cs
using Microsoft.Xna.Framework;
using Microsoft.Xna.Framework.Graphics;
using Microsoft.Xna.Framework.Input;
using System.IO;
using Inignoto.Utilities;
using System.Collections.Generic;

namespace Inignoto.Graphics.Textures
{
    public class Textures
    {
        public static TextureAtlas tiles;
        public static Texture2D fp_cursor;
        public static Texture2D hud;
        public static Texture2D viginette;

        private static List<Texture> textures = new List<Texture>();

        public static void LoadTextures()
        {
            tiles = new TextureAtlas(new ResourcePath("Inignoto", "textures/tiles", "assets"));
            fp_cursor = LoadTexture(new ResourcePath("Inignoto", "textures/gui/fp_cursor.png", "assets"));
            hud = LoadTexture(new ResourcePath("Inignoto", "textures/gui/hud.png", "assets"));
            viginette = LoadTexture(new ResourcePath("Inignoto", "textures/gui/viginette.png", "assets"));

        }

        public static Texture2D LoadTexture(ResourcePath path)
        {
            FileStream stream = FileUtils.GetStreamForPath(path, FileMode.Open);
            Texture2D texture = Texture2D.FromStream(Inignoto.game.GraphicsDevice, stream);
            stream.Close();
            textures.Add(texture);
            return texture;
        }

        public static void Dispose()
        {
            foreach (Texture tex in textures)
            {
                tex.Dispose();
            }
        }
    }
}
